compiler/resolution/preFold.cpp

```diff
@@ -3055,6 +3057,7 @@ static Type* createOrFindFunTypeFromAnnotation(AList& argList,
                                                   parent,
                                                   argList,
                                                   false,
+                                                  RET_VALUE,
```

Comment:
I do not understand this code well enough to see the ramifications of this default. I am willing to accept it for now.

Reply:
I made the new `retTag` argument required so that if/when future calls to the function are added in the future, people would think about what the return tag should be rather than having bugs like this lurk beneath the surface.  My understanding is that RET_VALUE is the default retTag for a function (see https://github.com/chapel-lang/chapel/blob/main/compiler/AST/FnSymbol.cpp#L55), so is what these calls had been getting before this change.  So here, I'm simply making that choice of default more explicit.  I looked briefly at this function to see whether there's something smarter we could/should be doing, but nothing jumped out at me, and so I concluded that I wasn't leaving things in any worse shape than they had been in for this call.